src/app/CommandSender.h

```diff
@@ -124,8 +151,30 @@ class CommandSender final : public Messaging::ExchangeDelegate
     CommandSender(Callback * apCallback, Messaging::ExchangeManager * apExchangeMgr, bool aIsTimedRequest = false,
                   bool aSuppressResponse = false);
     ~CommandSender();
-    CHIP_ERROR PrepareCommand(const CommandPathParams & aCommandPathParams, bool aStartDataStruct = true);
-    CHIP_ERROR FinishCommand(bool aEndDataStruct = true);
+
+    /**
+     * Enables CommandSender to send batch commands up until the number of supported by remote peer.
+     *
+     * It ensures that commands contain all required elements while building such that sending the
+     * InvokeRequestMessage has a high likeihood to succeed. Must be called before PrepareCommand.
+     * If aRemoteMaxPathsPerInvoke is 1, this will allow the call to contain only one command and
+     * doesn't enforce other batch commands requirements (similar to have never calling this method.
+     *
+     * @param [in] aRemoteMaxPathsPerInvoke max number of paths per invoke supported by remote peer.
+     *
+     * @return CHIP_ERROR_INCORRECT_STATE
+     *                      If device has previously called `PrepareCommand`.
+     * @return CHIP_ERROR_INVALID_ARGUMENT
+     *                      Invalid argument value.
+     * @return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE
+     *                      Device has not enabled CHIP_CONFIG_SENDING_BATCH_COMMANDS_ENABLED.
+     */
+    CHIP_ERROR SetBatchCommandsConfig(uint16_t aRemoteMaxPathsPerInvoke);
```

Comment:
If this is named `Set...Config` should it take a struct of `const ConfigParams&`? or if not, alternatively this should be `SetBatchCommandsMaxPaths` since this is what it does.

Thinking that if we ever add more config, generally numbers and bools are hard to read anyway (like `SetBatchCommandsConfig(10, true, 1024)` would be hard to figure out) so we should either make it a struct for readability or claim we will not have many configs for now and name this explicit.